### ISSUE TITLE
[examples/gatsbygram] Adding a page inside a modal

### ISSUE THREAD

**Issue by daydream05** [gatsbyjs/gatsby#4894]
In [gatsbygram](https://github.com/gatsbyjs/gatsby/tree/master/examples/gatsbygram), I noticed that the way you insert a page into a modal is through the layout. And in this layout, you check if the page' pathname is not any of the following. 

```
    let isModal = false

    if (
      this.props.location.pathname !== `/` &&
      this.props.location.pathname !== `/about/` &&
      this.windowWidth > 750
    ) {
      isModal = true
    }
```

However, as the app grows there would be a lot more pathnames and it would be cumbersome to add every single one of them. Is there a better way to specify if a page should be contained inside a modal? I know this might be more of a general programming/react question and less of Gatsby but I'm struggling to find a way to have the layout know about the page's property aside from their pathname. I was thinking of using boundActionCreators or using React's context to pass any data back to the layout but I'm lost. Thanks

**Comment by KyleAMathews** [comment]
Where would you like the source of truth about whether a page should be in a modal or not?

**Comment by daydream05** [comment]
If i'm going to have multiple pages that would likely use a modal. (i.e. multiple album pages), I was thinking of housing the source of truth on the page that activated the modal. So then when i close the modal, it would go back to this page without having to hardcode that page's link to the  modal component.

**Comment by daydream05** [comment]
I'm going to close this one. I figured out a way by using the [location](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/location.md) object! I realized that when to access a page contained-modal we're essentially linking to the page. 

`<Link
        to={{
          pathname: slug,
          state: { 
            isInModal: true,
         }
       }}
>`

Now that the page(including the layout) has access to the location's state, I can just check if I passed a state to open it in a modal.

`if (this.props.location.state) {
  if (
    this.props.location.state.isInModal  &&
    this.windowWidth > 750
  ) {
    isModal = true;
  }
}`
